src/lib/gadgets/gadgets.unit-test.ts

```diff
@@ -116,3 +117,115 @@ test(
     });
   }
 );
+
+// --------------------------
+// Shift Gates
+// --------------------------
+
+function testShift(
+  field: Field,
+  bits: number,
+  mode: 'left' | 'right',
+  result: Field
+) {
+  Provable.runAndCheck(() => {
+    let w = Provable.witness(Field, () => field);
+    let r = Provable.witness(Field, () => result);
+    let output = Provable.if(
+      Bool(mode === 'left'),
+      Gadgets.leftShift(w, bits),
+      Gadgets.rightShift(w, bits)
+    );
+    output.assertEquals(
+      r,
+      `${mode === 'left' ? 'ls' : 'rs'}(${field}, ${bits})`
+    );
+  });
+}
+
+let LS = ZkProgram({
+  methods: {
+    run: {
+      privateInputs: [Field],
+      method(x) {
+        Gadgets.leftShift(x, 2);
+      },
+    },
+  },
+});
+
+await LS.compile();
+await equivalentAsync({ from: [maybeUint64], to: boolean }, { runs: 3 })(
+  (x) => {
+    if (x >= 1n << 64n) throw Error('expected 64 bits');
+    return true;
```

Comment:
Instead of just returning true here, we can also make the `ZkProgram` return the output of the shift operation and make sure that its equivalent to the out of circuit variant not just in generating a valid proof and catching errors but also the correctness of the actual output

Reply:
Fixed in https://github.com/o1-labs/o1js/pull/1194/commits/b7e427c24a0396556450836a18302814a511ee74